### PR TITLE
New weather and time system

### LINKED ISSUE #888: Change gmaps api to HERE api
do it asap and we not using fucking bing maps

### PR DESCRIPTION
**Pull request by Doges**
yeah, i don't like google maps
Closes #888

### CODE REVIEW

File: src/apis/PositionStack.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      envVars: ['PS_ACCESS_KEY']
+    })
+  }
+
+  async getAddress (city) {
+    return this.request('forward', city).then(res => res.data)
+  }
+
+  request (endpoint, query = '') {
+    return axios.get(encodeURI(`${API_URL}/${endpoint}?query=${query}&access_key=${process.env.PS_ACCESS_KEY}&limit=1&output=json&timezone_module=1`))
```
Comment (almeidx) [inline src/apis/PositionStack.js:19]: perhaps use the options.params to make the url more readable
maybe even baseURL

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/apis/PositionStack.js (hunk 1 above)
Comment (Doges) [reply src/apis/PositionStack.js:19]: there is one endpoint, can make that more readable, 1s

File: src/commands/utility/weather.js [hunk 2]
```diff
@@ -22,13 +22,12 @@ module.exports = class Weather extends Command {
   async run ({ t, author, channel, language }, address) {
     moment.locale(language)
     channel.startTyping()
-    const city = await this.client.apis.gmaps.searchCity(address)
-    if (city) {
-      const [lang] = language.split('-')
-      const { lat, lng } = city.geometry.location
+    const city = await this.client.apis.positionstack.getAddress(address)
```
Comment (moruzerinho6) [inline src/commands/utility/weather.js:25]: Not gonna handle errors?

Comment (Doges) [reply src/commands/utility/weather.js:25]: whoops, forgot about it

File: src/commands/utility/weather.js [hunk 3]
```diff
@@ -45,14 +44,11 @@ module.exports = class Weather extends Command {
...
-      const cityName = city.address_components.find(({ types }) => types.includes('administrative_area_level_2') || types.includes('locality')).short_name
-      const state = city.address_components.find(({ types }) => types.includes('administrative_area_level_1'))
-
       const tempUnit = '°C'
-      const weather = await CanvasTemplates.weather({ t }, `${cityName.toUpperCase()}${state ? ` - ${state.short_name}` : ''}`, weatherInfo, tempUnit)
+      const weather = await CanvasTemplates.weather({ t }, `${city.data[0].county.toUpperCase()}${city.data[0].region ? ` - ${city.data[0].region_code}` : ''}`, weatherInfo, tempUnit)
 
       channel.send(new MessageAttachment(weather, 'weather.png')).then(() => channel.stopTyping())
-    } else {
+    } catch (e) {
```
Comment (almeidx) [inline src/commands/utility/weather.js:51]: Unused variable
```suggestion
    } catch {
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]
LGTM

**Review by moruzerinho6 — APPROVED** [review]

**Pull request merged**